pkg/spec/resources.go

```diff
@@ -585,6 +587,22 @@ func (app *ControllerFields) CreateK8sObjects() ([]runtime.Object, []string, err
 	objects = append(objects, configMap...)
 	log.Debugf("app: %s, configMap: %s\n", app.Name, spew.Sprint(configMap))
 
+	//Adding Annotations to the metadata of all the objects based on their type
+	if app.Appversion != "" {
+		accessor := meta.NewAccessor()
+		for _, object := range objects {
+			annotations, err := accessor.Annotations(object)
+			if err != nil {
+				return nil, nil, errors.Wrap(err, "cannot get annotations")
+			}
+			annotations = addKeyValueToMap(appVersion, app.Appversion, annotations)
+			err = accessor.SetAnnotations(object, annotations)
+			if err != nil {
+				return nil, nil, errors.Wrap(err, "cannot set annotations")
+			}
+		}
+	}
```

Comment:
this needs to move to [`CoreOperations`](https://github.com/kedgeproject/kedge/blob/cd5297cfdbd2f1daa510824d49c9d3d649ffc0b8/pkg/spec/controller.go#L98) right ?

Also you can put it in a function so, there won't be a need to have this code in between!

Reply:
@surajssd I tried it there but I was not able to get the value of appversion there, can you please suggest me some method to do that. 

One I thought was that we can return the value of appversion along with all runtimeObjects